fanout.go

```diff
@@ -124,21 +108,68 @@ func (f *Fanout) ServeDNS(ctx context.Context, w dns.ResponseWriter, m *dns.Msg)
 	return 0, nil
 }
 
+type clientSelector interface {
+	Pick() Client
+}
+
+func (f *Fanout) runWorkers(ctx context.Context, req *request.Request) chan *response {
+	var sel clientSelector
+	if f.serverCount == len(f.clients) {
+		sel = selector.NewSimpleSelector(f.clients)
+	} else {
+		sel = selector.NewWeightedRandSelector(f.clients, f.loadFactor)
+	}
+
+	workerChannel := make(chan Client, f.workerCount)
```

Comment:
[minor] to be consistent with the next line.
```suggestion
	workerCh := make(chan Client, f.workerCount)
```

Reply:
Done